go.mod

```diff
@@ -4,60 +4,60 @@ go 1.22.0
 
 require (
 	github.com/aws/aws-lambda-go v1.47.0
-	github.com/aws/aws-sdk-go v1.54.8
-	github.com/aws/aws-sdk-go-v2 v1.30.4
-	github.com/aws/aws-sdk-go-v2/config v1.27.30
-	github.com/aws/aws-sdk-go-v2/credentials v1.17.29
-	github.com/aws/aws-sdk-go-v2/feature/dynamodb/attributevalue v1.14.12
-	github.com/aws/aws-sdk-go-v2/feature/dynamodb/expression v1.7.34
-	github.com/aws/aws-sdk-go-v2/feature/s3/manager v1.17.14
-	github.com/aws/aws-sdk-go-v2/service/cloudwatch v1.40.5
-	github.com/aws/aws-sdk-go-v2/service/dynamodb v1.34.6
-	github.com/aws/aws-sdk-go-v2/service/eventbridge v1.33.3
-	github.com/aws/aws-sdk-go-v2/service/firehose v1.32.2
-	github.com/aws/aws-sdk-go-v2/service/kinesis v1.29.5
-	github.com/aws/aws-sdk-go-v2/service/lambda v1.58.1
-	github.com/aws/aws-sdk-go-v2/service/s3 v1.60.1
-	github.com/aws/aws-sdk-go-v2/service/secretsmanager v1.32.6
-	github.com/aws/aws-sdk-go-v2/service/sns v1.31.5
-	github.com/aws/aws-sdk-go-v2/service/sqs v1.34.5
-	github.com/aws/aws-sdk-go-v2/service/sts v1.30.5
+	github.com/aws/aws-sdk-go v1.55.5
+	github.com/aws/aws-sdk-go-v2 v1.32.2
+	github.com/aws/aws-sdk-go-v2/config v1.28.0
+	github.com/aws/aws-sdk-go-v2/credentials v1.17.41
+	github.com/aws/aws-sdk-go-v2/feature/dynamodb/attributevalue v1.15.12
+	github.com/aws/aws-sdk-go-v2/feature/dynamodb/expression v1.7.47
+	github.com/aws/aws-sdk-go-v2/feature/s3/manager v1.17.33
+	github.com/aws/aws-sdk-go-v2/service/cloudwatch v1.42.2
+	github.com/aws/aws-sdk-go-v2/service/dynamodb v1.36.2
+	github.com/aws/aws-sdk-go-v2/service/eventbridge v1.35.2
+	github.com/aws/aws-sdk-go-v2/service/firehose v1.34.2
+	github.com/aws/aws-sdk-go-v2/service/kinesis v1.32.2
+	github.com/aws/aws-sdk-go-v2/service/lambda v1.63.2
+	github.com/aws/aws-sdk-go-v2/service/s3 v1.66.0
+	github.com/aws/aws-sdk-go-v2/service/secretsmanager v1.34.2
+	github.com/aws/aws-sdk-go-v2/service/sns v1.33.2
+	github.com/aws/aws-sdk-go-v2/service/sqs v1.36.2
+	github.com/aws/aws-sdk-go-v2/service/sts v1.32.2
 	github.com/aws/aws-xray-sdk-go v1.8.4
-	github.com/awslabs/kinesis-aggregation/go/v2 v2.0.0-20230808105340-e631fe742486
+	github.com/awslabs/kinesis-aggregation/go/v2 v2.0.0-20241004223953-c2774b1ab29b
 	github.com/golang/protobuf v1.5.4
 	github.com/google/go-jsonnet v0.20.0
 	github.com/google/uuid v1.6.0
 	github.com/hashicorp/go-retryablehttp v0.7.7
 	github.com/iancoleman/strcase v0.3.0
 	github.com/itchyny/gojq v0.12.16
-	github.com/klauspost/compress v1.17.9
-	github.com/oschwald/maxminddb-golang v1.13.0
+	github.com/klauspost/compress v1.17.11
+	github.com/oschwald/maxminddb-golang v1.13.1
 	github.com/sirupsen/logrus v1.9.3
 	github.com/spf13/cobra v1.8.1
-	github.com/tidwall/gjson v1.17.1
+	github.com/tidwall/gjson v1.18.0 // Upgrades require SemVer bump.
```

Comment:
I understand the intent that a upgrade to new features in `gjson` can give users new features and should be considered a features, but this specific upgrade to `v1.18` seems to be a [performance](https://github.com/tidwall/gjson/releases/tag/v1.18.0) improvement. Since this isn't necessarily a Substation feature we might not need a version bump for this specific upgrade?

Reply:
I think the tldr on this is that we shouldn't make assumptions about _why_ a dependency bumped their version they way that they did -- they know their code better than we do. It's low effort (and not very impactful on users) for us to bump our own version whenever these packages do.

In this case, the changes look like it should have been a patch release, but for some reason it's not and I don't think we need to put much thought into why and how it affects our own versioning. (The caveat to this is if they break a unit test with a non-breaking update, then that's different and we should be more thoughtful about upgrading.)